state/protocol/badger/state.go

```diff
@@ -781,7 +781,11 @@ func (state *State) populateCache() error {
 		// sealed root height
 		err = state.db.View(operation.RetrieveSealedRootHeight(&state.sealedRootHeight))
 		if err != nil {
-			return fmt.Errorf("could not read sealed root block to populate cache: %w", err)
+			if errors.Is(err, storage.ErrNotFound) {
```

Comment:
When we extract snapshot from an v0.30 EN which doesn't have the sealedRootHeight stored in badger, it will return key not found error. In order to be backward compatible, we handle this error and use the finalizedRootHeight as the sesaledRootHeight.

This change is only needed for v0.31 below (which are v0.30 for existing devnet and v0.29 for existing mainnet), because after sporking with v0.31, all nodes should have sealedRootHeight stored in badger.